lib/ansible/modules/network/avi/avi_pool.py

```diff
@@ -0,0 +1,338 @@
+        default: 1
+    lb_algorithm_consistent_hash_hdr:
+        description:
+            - Http header name to be used for the hash key. 
+    lb_algorithm_hash:
+        description:
+            - Criteria used as a key for determining the hash between the client and  server. 
+        default: 1
+    max_concurrent_connections_per_server:
+        description:
+            - The maximum number of concurrent connections allowed to each server within the pool. 
+        default: 0
+    max_conn_rate_per_server:
+        description:
+            - Rate limit connections to each server. 
+    name:
+        description:
+            - The name of the pool. 
+        required: true
+    networks:
+        description:
+            - (internal-use) networks designated as containing servers for this pool. The servers may be further narrowed down by a filter. This field is used internally by avi, not editable by the user. 
+    pki_profile_ref:
+        description:
+            - Avi will validate the ssl certificate present by a server against the selected pki profile. It is a reference to an object of type pkiprofile. 
+    placement_networks:
+        description:
+            - Manually select the networks and subnets used to provide reachability to the pool's servers. Specify the subnet using the following syntax  10. 1. 1. 0/24. If the pool servers are not directly connected, but routable from the serviceengine, please also provide the appropriate static routes to reach the servers in the vrf configuration. 
+    prst_hdr_name:
+        description:
+            - Header name for custom header persistence. 
+    request_queue_depth:
+        description:
+            - Minimum number of requests to be queued when pool is full. 
+        default: 128
+    request_queue_enabled:
+        description:
+            - Enable request queue when pool is full. 
+        default: False
+    rewrite_host_header_to_server_name:
+        description:
+            - Rewrite incoming host header to server name of the server to which the request is proxied. Enabling this feature rewrites host header for requests to all servers in the pool. 
+        default: False
+    rewrite_host_header_to_sni:
+        description:
+            - If sni server name is specified, rewrite incoming host header to the sni server name. 
+        default: False
+    server_auto_scale:
+        description:
+            - Server autoscale. Not used anymore. 
+        default: False
+    server_count:
+        description:
+            - Number of server_count. 
+        default: 0
+    server_name:
+        description:
+            - Fully qualified dns hostname which will be used in the tls sni extension in server connections if sni is enabled. If no value is specified, avi will use the incoming host header instead. 
+    server_reselect:
+        description:
+            - Server reselect configuration for http requests. 
+    servers:
+        description:
```

Comment:
some of the options dont have very helpful descriptions:

'A list of servers for the pool to direct trafic to' ... also a note to see the format of the server in the examples would be helpful

used this one as an example, but several of the descriptions could be made clearer